app/code/core/Mage/Catalog/Model/Resource/Product/Compare/Item/Collection.php

```diff
@@ -236,13 +234,18 @@ public function getComparableAttributes()
                     ->joinLeft(
                         array('al' => $this->getTable('eav/attribute_label')),
                         'al.attribute_id = main_table.attribute_id AND al.store_id = ' . (int) $this->getStoreId(),
-                        array('store_label' => $this->getConnection()->getCheckSql('al.value IS NULL', 'main_table.frontend_label', 'al.value'))
+                        array('store_label' => new Zend_Db_Expr('IFNULL(al.value, main_table.frontend_label)'))
+                    )
+                    ->joinLeft(
+                        array('ai' => $this->getTable('eav/entity_attribute')), 
+                        'ai.attribute_id = main_table.attribute_id'
                     )
                     ->where('additional_table.is_comparable=?', 1)
-                    ->where('main_table.attribute_id IN(?)', $attributeIds);
+                    ->where('ai.attribute_set_id IN(?)', $setIds)
+                    ->order(array('ai.attribute_group_id ASC', 'ai.sort_order ASC'));
                 $attributesData = $this->getConnection()->fetchAll($select);
                 if ($attributesData) {
-                    $entityType = Mage_Catalog_Model_Product::ENTITY;
+                    $entityType = 'catalog_product';
```

Comment:
What is the purpose of ceasing to use a constant here?

Reply:
I borrowed the code here https://magento.stackexchange.com/questions/64983/attributes-order-in-comparison-window-is-wrong. I did a PR to discuss it. I tested both changes and they do not bring any changes. I will modify the code to keep the initial version. Although the code has been working for years without any issues, what I would be interested in is to see if the change has the format you are taught. Also, if the database query is optimized.